lib/http-delegations.js

```diff
@@ -0,0 +1,97 @@
+/*!
+ * Copyright (c) 2019-2020 Digital Bazaar, Inc. All rights reserved.
+ */
+'use strict';
+
+const {asyncHandler} = require('bedrock-express');
+const bedrock = require('bedrock');
+const {config, util: {BedrockError}} = bedrock;
+const brPassport = require('bedrock-passport');
+const brZCapStorage = require('bedrock-zcap-storage');
+const {ensureAuthenticated} = brPassport;
+
+bedrock.events.on('bedrock-express.configure.routes', app => {
+  const cfg = config['zcap-storage-http'];
+  const {routes} = cfg;
+
+  // insert a delegation
+  app.post(
+    routes.delegations,
+    ensureAuthenticated,
+    // TODO: add validator
+    //validate('bedrock-zcap-storage-http.zcap'),
+    asyncHandler(async (req, res) => {
+      // TODO: consider invoking the `account` as a zcap here ... and add
+      // `invoker` and `delegator` to the account to avoid having a permission
+      // based model here; automatically add an admin account as an invoker
+      // whenever the account is retrieved that can't be removed
+      const {actor} = req.user;
+      // if controller is not specified it will be the authenticated user
+      const {controller = actor.id, capability, delegator, domain} = req.body;
+      // const referenceId = req.body.referenceId || capability.referenceId;
+      if(actor.id !== controller) {
```

Comment:
JFYI - Reviewing this is blocked on design considerations around `controller` here and whether not these delegations should instead be stored in an EDV.

Reply:
@dlongley have decisions been made?  How does that affect this module/PR?